pkg/resource/deploy/step_generator.go

```diff
@@ -307,7 +303,7 @@ func (sg *stepGenerator) GenerateSteps(event RegisterResourceEvent) ([]Step, res
 		}
 	}
 
-	if !sg.isTargetedUpdate() {
+	if !sg.updateTargetsOpt.IsConstrained() {
 		return steps, nil
 	}
```

Comment:
This used to early exit for updates where `--replace` was specified...

This is a behavior change. Consider the repercussions.

Reply:
I've removed the behavior change to avoid potential regressions.